code/modules/paperwork/clipboard.dm

```diff
@@ -33,156 +35,182 @@
 			return TRUE
 	. = ..()
 
+/obj/item/clipboard/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	if(stored_pen)
+		to_chat(user, "It's holding \a [stored_pen].")
+	if(!LAZYLEN(papers))
+		to_chat(user, "It contains [length(papers)] / [max_papers] paper(s).")
+	else 
+		to_chat(user, "It has room for [max_papers] paper(s).")
+
+/obj/item/clipboard/proc/top_paper()
+	return LAZYACCESS(papers, 1)
+
+/obj/item/clipboard/proc/push_paper(var/obj/item/P)
+	papers = papers? list(P) + papers : list(P) //Push at the top
+
+/obj/item/clipboard/proc/pop_paper()
+	. = top_paper()
+	LAZYREMOVE(papers, 1)
+
 /obj/item/clipboard/on_update_icon()
 	..()
-	if(toppaper)
-		overlays += overlay_image(toppaper.icon, toppaper.icon_state, flags=RESET_COLOR)
-		overlays += toppaper.overlays
-	if(haspen)
+	var/obj/item/top_paper = top_paper()
+	if(top_paper)
+		overlays += overlay_image(top_paper.icon, top_paper.icon_state, flags=RESET_COLOR)
```

Comment:
Would be nice if we could refactor overlays here to use SSoverlays but I appreciate that would mean refactoring every /obj/item that uses ..()

Reply:
I'm not sure I follow?

Reply:
Mostly just waffling, disregard.